config/initializers/highlight_override.rb

```diff
@@ -0,0 +1,31 @@
+# Monkey patch the generated Bindings for the swagger Highlight model
+require_relative '../../app/bindings/api/v0/bindings/new_highlight'
+
+module Api::V0::Bindings
+  class NewHighlight
+    def valid_location_strategies?
+      location_strategies = to_hash[:location_strategies]
+      location_strategies.all? do |location_strategy|
+        strategy = JSON.parse(location_strategy)
+        case strategy['type']
+        when 'TextPositionSelector'
+          keys_matcher(%w(start end), strategy.keys)
+        when 'XpathRangeSelector'
+          keys_matcher(%w(endContainer endOffset startContainer startOffset),
+                       strategy.keys)
+        else
```

Comment:
Instead of validating by checking for keys, was thinking we'd define schemas for `TextPositionSelector` and `XpathRangeSelector` and generate bindings for them.  This way we don't have to hard code properties in the monkey patch and we get a more full validity check and error reporting.

Then overriding `location_strategies=`:

```ruby
def location_strategies=(array)
  @location_strategies = array.map do |item|
    case item[:type]
    when 'TextPositionSelector'
      Api::V0::Bindings::TextPositionSelector.new(item)
    when 'XpathRangeSelector'
      Api::V0::Bindings::XpathRangeSelector.new(item)
    end
  end
end
```

Then `valid?` can delegate to the `valid?` method on the strategy bindings:

```ruby
def valid?
  old_valid? && !location_strategies.empty? && location_strategies.all?(&:valid?)
end
```

Additionally `list_invalid_properties` (used in bind to report errors) can be extended:

```ruby
alias_method :old_list_invalid_properties, :list_invalid_properties
def list_invalid_properties
  invalid_properties = old_list_invalid_properties

  if location_strategies.empty?
    invalid_properties.push('invalid value for "location_strategies", location_strategies cannot be empty.')
  end

  location_strategies.each do |strategy|
    strategy.list_invalid_properties.each do |strategy_invalid_property|
      invalid_properties.push("invalid value for location strategy #{strategy.type}: #{strategy_invalid_property}")
    end
  end 

  invalid_properties
end
```